device/include/DataDescriptor.h

```diff
@@ -89,6 +89,12 @@ namespace ChimeraTK {
      */
     [[nodiscard]] DataType rawDataType() const;
 
+    /**
+     * Set the raw data type. This is useful e.g. when a decorated register should no longer allow
+     * raw access, in which case you should set DataType::none
+     */
+    void setRawDataType(const DataType& d);
```

Comment:
Constructing a new DataDescriptor from a given one requires a lot of logic since you need to test many conditions before you get the required value from the given DataDescriptor. That's why we decided to modify a copy, using a setter.